### PR TITLE
bug/PDPDEVTOOL-3604-escape-in-setupaccount

### PR DESCRIPTION
**Pull request by x8a**
- Fixed the problem of the escape key not canceling the command
- Edited the message
- Added the NetSuite domain constant

### CODE REVIEW

File: packages/vscode-extension/messages.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	"COMMAND_SUCCESS": "The {0} command has been successfully executed.",
 	"COMMAND_TRIGGERED": "{0} has been triggered.",
 	"DISMISS": "Dismiss",
+	"PRODUCTION_DOMAIN": "system.netsuite.com",
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/messages.json:8]: You can reuse the ApplicationConstants.PROD_ENVIRONMENT_ADDRESS from packages/nodecli. In vscode, ApplicationConstants is accessed via ExtensionUtil.

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 2]
```diff
@@ -141,14 +141,17 @@ export default class ManageAccounts extends BaseAction {
 
 	private async handleBrowserAuth(accountCredentialsList: AuthListData) {
 		const authId = await this.getNewAuthId(accountCredentialsList);
-		const url = await this.getUrl();
-		if (!authId) {
+		if (!authId || authId === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:144]: !authId already checks for null, undefined, and empty string. So for this one, I think !authId should be enough.

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 3]
```diff
@@ -223,9 +226,11 @@ export default class ManageAccounts extends BaseAction {
 			placeHolder: this.translationService.getMessage(MANAGE_ACCOUNTS.CREATE.ENTER_URL),
 			ignoreFocusOut: true,
 			validateInput: (fieldValue) => {
+				if(!fieldValue) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:229]: Please add a space after if. That's what we're using as standard for our code.

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 4]
```diff
@@ -281,11 +286,27 @@ export default class ManageAccounts extends BaseAction {
 
 	private async handleSaveToken(accountCredentialsList: AuthListData) {
 		const authId = await this.getNewAuthId(accountCredentialsList);
+		if(!authId || authId === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:289]: I think !authId is enough here. No need to add authId === undefined.

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 5]
```diff
@@ -281,11 +286,27 @@ export default class ManageAccounts extends BaseAction {
...
+			return;
+		}
+
 		const accountId = await this.getAccountId();
+		if(!accountId || accountId === undefined) {
+			return;
+		}
+
 		const tokenId = await this.getTokenId();
+		if(!tokenId || tokenId === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:304]: I think !tokenId should be enough here.

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 6]
```diff
@@ -281,11 +286,27 @@ export default class ManageAccounts extends BaseAction {
...
+			return;
+		}
+
 		const tokenSecret = await this.getTokenSecret();
-		if (!authId || !accountId || !tokenId || !tokenSecret) {
+		if(!tokenSecret || tokenSecret === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:309]: I think !tokenSecret should be enough here.

File: packages/vscode-extension/src/service/TranslationKeys.ts [hunk 7]
```diff
@@ -97,3 +97,4 @@ export const DISMISS = 'DISMISS';
 export const SEE_DETAILS = 'SEE_DETAILS';
 export const YES = 'YES';
 export const NO = 'NO';
+export const PRODUCTION_DOMAIN = 'PRODUCTION_DOMAIN';
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/service/TranslationKeys.ts:100]: Please check my comment in ManageAccounts.ts.

**Review by karl-anthony-ng — CHANGES_REQUESTED** [review]

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 8]
```diff
@@ -141,14 +141,17 @@ export default class ManageAccounts extends BaseAction {
...
-		const url = await this.getUrl();
 		if (!authId) {
 			return;
 		}
+		const url = await this.getUrl();
 		const commandParams: { authid: string; dev: boolean; url?: string } = {
 			authid: authId,
 			dev: false,
 		};
+		if (url === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:152]: Can we use !url here?

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 9]
```diff
@@ -64,7 +64,7 @@ export default class ManageAccounts extends BaseAction {
 
 		if (actionResult.status === actionResultStatus.SUCCESS) {
 			const selected = await this.getAuthListOption(actionResult.data);
-			if (!selected) {
+			if (!selected || selected === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:67]: Can we use just !selected here?

File: packages/vscode-extension/src/commands/ManageAccounts.ts [hunk 10]
```diff
@@ -281,13 +286,30 @@ export default class ManageAccounts extends BaseAction {
 
 	private async handleSaveToken(accountCredentialsList: AuthListData) {
 		const authId = await this.getNewAuthId(accountCredentialsList);
+		if (!authId) {
+			return;
+		}
+
 		const url = await this.getUrl();
+		if (url === undefined) {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/commands/ManageAccounts.ts:294]: Can we use !url here?

**Review by karl-anthony-ng — CHANGES_REQUESTED** [review]

File: packages/vscode-extension/src/commands/ManageAccounts.ts (hunk 8 above)
Comment (x8a) [reply packages/vscode-extension/src/commands/ManageAccounts.ts:152]: It didn't work.

File: packages/vscode-extension/src/commands/ManageAccounts.ts (hunk 10 above)
Comment (x8a) [reply packages/vscode-extension/src/commands/ManageAccounts.ts:294]: It didn't work either.

**Review by karl-anthony-ng — APPROVED** [review]

**Review by vic-nik — APPROVED** [review]

**Pull request merged**